src/steamship/base/client.py

```diff
@@ -227,14 +227,14 @@ def _url(
 
         return f"{base}/{operation}"
 
-    def _headers(
+    def _headers(  # noqa: C901
         self,
         is_package_call: bool = False,
         package_owner: str = None,
         package_id: str = None,
         package_instance_id: str = None,
         as_background_task: bool = False,
-        wait_on_tasks: List[Task] = None,
+        wait_on_tasks: List[Union[str, Task]] = None,
```

Comment:
I think it would be better if we only supported List[Task]. Don't know yet why we need to support task_id. Will come back to this.

Reply:
I didn't see any spots where we need to work with just task_id. So I would only support List[Task]

Reply:
Happy to collapse it, though a trivial example of where taskIds get passed around is the return value of some package methods